Production/src/MuonHitMatch.cc

```diff
@@ -72,6 +72,12 @@ void MuonHitMatch::CountHits(const pat::Muon& muon, CountMap& n_hits)
                         muon_n_hits = &n_hits.at(MuonSubdetId::CSC);
                     else if(hit_pattern.muonRPCHitFilter(hit_id))
                         muon_n_hits = &n_hits.at(MuonSubdetId::RPC);
+                    else if(hit_pattern.muonGEMHitFilter(hit_id))
+                        muon_n_hits = &n_hits.at(MuonSubdetId::GEM);
+                    else if(hit_pattern.muonME0HitFilter(hit_id)) {
+                        muon_n_hits = &n_hits.at(MuonSubdetId::ME0);
+                        station_index = GetStationIndex(1, false);
```

Comment:
could you, please, add some comment here to document why this fix is needed for ME0?

Reply:
Done

Reply:
Just for my understanding, why `hit_pattern.getMuonStation(hit_id)` fails to return the correct station for ME0 and works for other sub-detectors? Is it a known bug in CMSSW? Instead, `segment.station()` works fine when counting matches for ME0 (in `CountMatches`)?

Reply:
This is something I've just figured out by having a look at the outcome of these functions & what is then stored in the n_hits and n_matches quantities. Would need to ask & cross-check that on hypernews (I think tracking POG), why this is the case, to be honest.

I guess this is the case, since the tracking information (also for muon systems) seems to be still in development for Phase2, such that the new muonStations don't work properly, at least ME0

Reply:
Ok, thanks for the details! For this PR, let's keep the workaround as you implemented it. Then, after we have feedback from the tracking POG, we might come back to it.